lib/Dialect/Substrait/IR/Substrait.cpp

```diff
@@ -155,6 +155,22 @@ EmitOp::inferReturnTypes(MLIRContext *context, std::optional<Location> loc,
   return success();
 }
 
+LogicalResult
+FetchOp::inferReturnTypes(MLIRContext *context, std::optional<Location> loc,
```

Comment:
It may be possible to remove this function and do type inference declaratively. See comment in `SubstraitOps.td`.